src/docquery/ocr_processor.py

```diff
@@ -0,0 +1,152 @@
+
+        # turn coordinates into (left, top, left+width, top+height) format
+        actual_boxes = [[x, y, x + w, y + h] for x, y, w, h in zip(left, top, width, height)]
+
+        image_width, image_height = image.size
+
+        # finally, normalize the bounding boxes
+        normalized_boxes = [self.normalize_box(box, image_width, image_height) for box in actual_boxes]
+
+        assert len(words) == len(normalized_boxes), "Not as many words as there are bounding boxes"
+
+        return words, normalized_boxes
+
+    @staticmethod
+    def check_if_available():
+        if not TESSERACT_AVAILABLE:
+            raise NoOCRReaderFound(
+                "Unable to use pytesseract (OCR will be unavailable). Install tesseract to process images with OCR."
+            )
+
+
+class EasyOCRProcessor(OCRProcessor):
+    def __init__(self):
+        super().__init__()
+        self.reader = None
+
+    def apply_ocr(self, image: "Image.Image") -> tuple[list[Any], list[list[int]]]:
+        """Applies Easy OCR on a document image, and returns recognized words + normalized bounding boxes."""
+        if not self.reader:
+            # TODO: expose language currently setting to english
+            self.reader = easyocr.Reader(['en'], gpu=self.device > -1)
+
+        # apply OCR
+        data = self.reader.readtext(np.array(image))
+        boxes, words, acc = list(map(list, zip(*data)))
+
+        # filter empty words and corresponding coordinates
+        irrelevant_indices = set(idx for idx, word in enumerate(words) if not word.strip())
+        words = [word for idx, word in enumerate(words) if idx not in irrelevant_indices]
+        boxes = [coords for idx, coords in enumerate(boxes) if idx not in irrelevant_indices]
+
+        # turn coordinates into (left, top, left+width, top+height) format
+        actual_boxes = [tl + br for tl, tr, br, bl in boxes]
+
+        image_width, image_height = image.size
+
+        # finally, normalize the bounding boxes
+        normalized_boxes = [self.normalize_box(box, image_width, image_height) for box in actual_boxes]
+
+        assert len(words) == len(normalized_boxes), "Not as many words as there are bounding boxes"
+
+        return words, normalized_boxes
+
+    @staticmethod
+    def check_if_available():
+        if not EASYOCR_AVAILABLE:
+            raise NoOCRReaderFound(
+                "Unable to use easyocr (OCR will be unavailable). Install easyocr to process images with OCR."
+            )
+
+
+class DummyProcessor(OCRProcessor):
+    def __init__(self):
+        super().__init__()
+        self.reader = None
+
+    def apply_ocr(self, image: "Image.Image") -> tuple[list[Any], list[list[int]]]:
+        raise NoOCRReaderFound("Unable to find any OCR engine and OCR extraction was requested")
+
+    @staticmethod
+    def check_if_available():
+        logging.warning("Unable to find OCR processor might not be needed")
```

Comment:
This warning message is a bit unclear to me. Maybe something like "Falling back to a dummy OCR processor since none were found."